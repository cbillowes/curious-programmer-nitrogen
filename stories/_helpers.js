import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page