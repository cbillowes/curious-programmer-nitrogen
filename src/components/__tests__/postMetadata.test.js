import React from 'react'
import PostMetadata from '../postMetadata'
import { getSnapshot } from './_helpers'

describe(`Post metadata`, () => {
  it(`should render`, () => {
    const component = (
      <PostMetadata
        date="2019-02-01T00:00:00.000Z"
        author="Clarice Bouwer"
        ttr={42}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })

  it(`should render without an author`, () => {
    const component = (
      <PostMetadata
        date="2019-02-01T00:00:00.000Z"
        ttr={42}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })
})
