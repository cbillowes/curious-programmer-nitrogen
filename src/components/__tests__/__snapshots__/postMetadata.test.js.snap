// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post metadata should render 1`] = `
<div
  className="metadata"
>
  posted on
   
  <time
    dateTime={"2019-02-01T00:00:00.000Z"}
  >
    Friday, 1 February 2019
  </time>
   
  <span>
    by Clarice Bouwer
  </span>
   
  <span>
    (Est. 42 minute read)
  </span>
</div>
`;

exports[`Post metadata should render without an author 1`] = `
<div
  className="metadata"
>
  posted on
   
  <time
    dateTime={"2019-02-01T00:00:00.000Z"}
  >
    Friday, 1 February 2019
  </time>
   
  <span>
    by Clarice Bouwer
  </span>
   
  <span>
    (Est. 42 minute read)
  </span>
</div>
`;
